src/components/LanguageSelection/LanguageSelection.js

```diff
@@ -0,0 +1,104 @@
+import React, { useContext, useState } from "react";
+import { ConfigContext } from "Context";
+import { useAsync } from "react-use";
+import Loader from "components/Loader";
+import { FormattedMessage, useIntl } from "react-intl";
+import { v4 as uuidv4 } from "uuid";
+import {
+  ButtonGroup,
+  Heading,
+  LinkButton,
+} from "@utrecht/component-library-react";
+import { get, put } from "../../api";
+import PropTypes from "prop-types";
+
+const propTypes = PropTypes.shape({
+  heading: PropTypes.string,
+  headingLevel: PropTypes.number,
+  onLanguageChange: PropTypes.func,
+});
+
+const LanguageSelection = (props) => {
+  // Hook uses
+  const config = useContext(ConfigContext);
+  const { locale } = useIntl();
+  const [current, setCurrent] = useState(locale);
+  const getAvailableLanguages = async () => {
+    const result = await get(`${config.baseUrl}i18n/info`);
+    setCurrent(result.current); // state change if current changed
+    return result.languages.map(({ code, name }) => ({
+      lang: code,
+      textContent: code.toUpperCase(),
+      label: name,
+      current: code === result.current,
+    }));
+  };
+  const {
+    loading,
+    value: items,
+    error,
+  } = useAsync(
+    getAvailableLanguages,
+    [config.baseUrl] // actually current too?!
```

Comment:
yeah, but it should rather be `locale` then and that's what we should compare against.

We will do the GET all again on language changes because that will change the API response `current`, but if that wouldn't be used in comparisons, we could split this in two hooks so that the GET call would only be called once :thinking: 

This is okay for now though.